src/main/java/edu/ucsb/cs156/happiercows/controllers/ChatMessageController.java

```diff
@@ -0,0 +1,167 @@
+package edu.ucsb.cs156.happiercows.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.extern.slf4j.Slf4j;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+
+import edu.ucsb.cs156.happiercows.entities.ChatMessage;
+import edu.ucsb.cs156.happiercows.repositories.ChatMessageRepository;
+
+import edu.ucsb.cs156.happiercows.entities.User;
+import edu.ucsb.cs156.happiercows.entities.UserCommons;
+import edu.ucsb.cs156.happiercows.repositories.UserCommonsRepository;
+
+import org.springframework.security.core.Authentication;
+
+
+import java.util.Optional;
+
+@Tag(name = "Chat Message")
+@RequestMapping("/api/chat")
+@RestController
+@Slf4j
+public class ChatMessageController extends ApiController{
+
+    @Autowired
+    private ChatMessageRepository chatMessageRepository;
+
+    @Autowired
+    private UserCommonsRepository userCommonsRepository;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @Operation(summary = "Get all chat messages", description = "Get all chat messages associated with a specific commons.")
+    @PreAuthorize("hasAnyRole('ROLE_USER', 'ROLE_ADMIN')")
+    @GetMapping("/get")
+    public ResponseEntity<Object> getChatMessages(@Parameter(description = "The id of the common") @RequestParam Long commonsId,
+                                            @Parameter(name="page") @RequestParam int page,
```

Comment:
Please make page and size optional, as users will not have the option to use them. This is incompatible with the MVP frontend

Reply:
I'll make it so that `page`and `size` are optional parameters that default to `0` and `10`, respectively.

Reply:
Ian... I think for a chat channel, if you don't have paging, that's not MVP.    We learned with the jobs interface, that you have something that grows quickly (which an active chat channel with users would), the stack bogs down really fast if you don't have paging in place.